packages/utils/src/account.ts

```diff
@@ -18,23 +22,79 @@ export enum MnemonicLanguages {
   spanish,
 }
 
-export function generateMnemonic(
+const bip39Wrapper: Bip39 = {
+  mnemonicToSeedSync: bip39.mnemonicToSeedSync,
+  mnemonicToSeed: bip39.mnemonicToSeed,
+  generateMnemonic: (
+    strength?: number,
+    rng?: (size: number, callback: (err: Error | null, buf: Buffer) => void) => void,
+    wordlist?: string[]
+  ): Promise<string> => {
+    return new Promise((resolve, reject) => {
+      strength = strength || 128
+      rng = rng || randomBytes
+
+      rng(strength / 8, (error, randomBytesBuffer) => {
+        if (error) {
+          reject(error)
+        } else {
+          resolve(bip39.entropyToMnemonic(randomBytesBuffer.toString('hex'), wordlist))
+        }
+      })
+    })
+  },
+  validateMnemonic: bip39.validateMnemonic,
+}
+
+interface Bip39 {
+  mnemonicToSeedSync: (mnemonic: string, password?: string) => Buffer
+  mnemonicToSeed: (mnemonic: string, password?: string) => Promise<Buffer>
+  generateMnemonic: (
+    strength?: number,
+    rng?: (size: number) => Buffer,
+    wordlist?: string[]
+  ) => Promise<string>
+  validateMnemonic: (mnemonic: string, wordlist?: string[]) => boolean
+}
+
+export async function generateMnemonic(
   strength: MnemonicStrength = MnemonicStrength.s256_24words,
-  language?: MnemonicLanguages
-): string {
-  return bip39.generateMnemonic(strength, undefined, getWordList(language))
+  language?: MnemonicLanguages,
+  bip39ToUse: Bip39 = bip39Wrapper
+): Promise<string> {
+  return bip39ToUse.generateMnemonic(strength, undefined, getWordList(language))
+}
+
+export function validateMnemonic(
+  mnemonic: string,
+  language?: MnemonicLanguages,
+  bip39ToUse: Bip39 = bip39Wrapper
+) {
+  return bip39ToUse.validateMnemonic(mnemonic, getWordList(language))
 }
 
-export function validateMnemonic(mnemonic: string, language?: MnemonicLanguages) {
-  return bip39.validateMnemonic(mnemonic, getWordList(language))
+export async function generateKeys(
+  mnemonic: string,
+  password?: string,
+  addressIndex: number = 0,
+  bip39ToUse: Bip39 = bip39Wrapper
+): Promise<{ privateKey: string; publicKey: string }> {
+  const seed = await bip39ToUse.mnemonicToSeed(mnemonic, password)
+  const node = bip32.fromSeed(seed)
+  const newNode = node.derivePath(`${CELO_DERIVATION_PATH_BASE}/${addressIndex}`)
+  return {
+    privateKey: newNode.privateKey!.toString('hex'),
```

Comment:
Is there a type problem or can the PK possibly be null? Maybe we should check it and throw if not?

Reply:
I checked the code. The only possible way that the privateKey is throwing null/undefined, it's if you are trying to generate the node from a publicKey.
And only two functions lead to that `bip32.fromBase58` and `bip32.fromPublicKeyLocal`
The, `bip.fromSeed` will return a valid node with a privateKey, and will avoid that error

Reply:
I could add a throw and a comment